### ISSUE TITLE
Add batch/non-interactive/force mode

### ISSUE THREAD

**Issue by Skrupellos** [pimutils/vdirsyncer#425]
Currently, when syncing a pair, `vdirsyncer` asks for confirmation prior creating a new collection in a storage.

It would be nice to have a command line argument like `--batch` or `--force`, which assures that `vdirsyncer` will never expect any input. If input is really really required, it should fail.

A use case would be a cron job, which mirrors a remote source to a local one (e.g. offline contacts).

**Comment by untitaker** [comment]
Instead of introducing several modes I'd like to have a guarantee specific to `sync` that it will never prompt for input.

Currently it automatically invokes `vdirsyncer discover` if collections have changed (`vdirsyncer discover` is the one that asks about creating a collection), maybe it should just fail quickly and tell the user to run `vdirsyncer discover` interactively?

**Comment by untitaker** [comment]
@Skrupellos Could you check whether #438 fits your needs closely enough?

**Comment by Skrupellos** [comment]
@untitaker No. By not discovering the remote storage, no new/deleted collection would be found. Therefore, no 1:1 mirror of the remote storage would be possible.

My use case is pretty simple (and common I guess):
A automated cron job / script creates a 1:1 copy / mirror / read-only sync of a remote storage (the calendars of some user). This can be used by some logic to compute something (print a lit of upcoming events, calculate the time you spend in meetings, ... xD).

**Comment by untitaker** [comment]
I see. However, you can already do:

    yes | vdirsyncer discover
    vdirsyncer sync

**Comment by untitaker** [comment]
FWIW I think that having an option to `discover` that specifically skips prompts for creating collections is useful since it offers (theoretically) finer-grained control. However, I don't think much of a batch mode since this is IMO too big of a hammer just for this. A batch mode would be more generic, at the expense of fine-grained control. I feel like the latter is more important.

So I would accept a parameter `--create-collections/--no-create-collections` that skips the prompts for just collection creation.

**Comment by madduck** [comment]
I erroneously filed https://github.com/pimutils/khal/issues/589 against khal about this.

**Comment by untitaker** [comment]
@madduck in your case it would work to write a wrapper script that checks if stdout is a tty or not, and runs different commands accordingly.

**Comment by madduck** [comment]
@untitaker well, sure, if `--batch` or the like exists. But please don't make me have a different configuration file for interactive and non-interactive mode, just so that I can set different conflictresolution methods.

**Comment by untitaker** [comment]
You misunderstood me. That wrapper script is then invoked by vdirsyncer instead of vimdiff

On 21 February 2017 05:26:12 CET, "martin f. krafft" <notifications@github.com> wrote:
>@untitaker well, sure, if `--batch` or the like exists. But please
>don't make me have a different configuration file for interactive and
>non-interactive mode, just so that I can set different
>conflictresolution methods.
>
>-- 
>You are receiving this because you were mentioned.
>Reply to this email directly or view it on GitHub:
>https://github.com/pimutils/vdirsyncer/issues/425#issuecomment-281242286

-- 
Sent from my Android device with K-9 Mail. Please excuse my brevity.

**Comment by madduck** [comment]
@untitaker Eh. busted.

Nevertheless, if that script now should fail in non-interactive mode, this isn't exactly what it should look like:

    error: Unknown error occured for madduck_calendar/calendar.ics: Command
    '['/home/madduck/.config/vdirsyncer/confres.sh',
    '/home/madduck/.tmp/vdirsyncer-conflict.v_lwhbr8/madduck_calendar_local',
    '/home/madduck/.tmp/vdirsyncer-conflict.v_lwhbr8/madduck_calendar_remote']'
    returned non-zero exit status 1

This should probably be handled better, no?

**Comment by untitaker** [comment]
Do you mean the fact that the list repr is printed, or something else?

**Comment by madduck** [comment]
It's not really an unknown error. But I don't know how this would be best handled, actually.

**Comment by untitaker** [comment]
Ah, yes that could be improved.

**Comment by madduck** [comment]
maybe the repr output is really level INFO, and you just print what the script writes to stderr and declare the task failed?

**Comment by untitaker** [comment]
I think this is doable without change of log level, in any case please open a new issue about this because it's not entirely related.

On 22 February 2017 04:13:26 CET, "martin f. krafft" <notifications@github.com> wrote:
>maybe the repr output is really level INFO, and you just print what the
>script writes to stderr and declare the task failed?
>
>-- 
>You are receiving this because you were mentioned.
>Reply to this email directly or view it on GitHub:
>https://github.com/pimutils/vdirsyncer/issues/425#issuecomment-281556248

-- 
Sent from my Android device with K-9 Mail. Please excuse my brevity.
